### PR TITLE
Traverse custom alias-like rules in the rust analyzer aspect

### PR DESCRIPTION
**Pull request by hlopko**

### CODE REVIEW

File: rust/private/rust_analyzer.bzl [hunk 1]
```diff
@@ -52,17 +52,21 @@ def _rust_analyzer_aspect_impl(target, ctx):
     if hasattr(ctx.rule.attr, "rustc_flags"):
         cfgs += [f[6:] for f in ctx.rule.attr.rustc_flags if f.startswith("--cfg ") or f.startswith("--cfg=")]
 
-    # Save BuildInfo if we find any (for build script output)
     build_info = None
-    for dep in ctx.rule.attr.deps:
-        if BuildInfo in dep:
-            build_info = dep[BuildInfo]
+    dep_infos = []
+    if hasattr(ctx.rule.attr, "proc_macro_deps"):
```
Comment (illicitonion) [inline rust/private/rust_analyzer.bzl:57]: This feels worth a comment - I'm assuming this is for sniffing whether we're a rules_rust rule vs some other one, which happens to also expose `BuildInfo`?

File: rust/private/rust_analyzer.bzl [hunk 2]
```diff
@@ -52,17 +52,21 @@ def _rust_analyzer_aspect_impl(target, ctx):
...
+                build_info = dep[BuildInfo]
+        dep_infos = [dep[RustAnalyzerInfo] for dep in ctx.rule.attr.deps if RustAnalyzerInfo in dep]
 
-    dep_infos = [dep[RustAnalyzerInfo] for dep in ctx.rule.attr.deps if RustAnalyzerInfo in dep]
     if hasattr(ctx.rule.attr, "proc_macro_deps"):
         dep_infos += [dep[RustAnalyzerInfo] for dep in ctx.rule.attr.proc_macro_deps if RustAnalyzerInfo in dep]
     if hasattr(ctx.rule.attr, "crate") and ctx.rule.attr.crate != None:
         dep_infos.append(ctx.rule.attr.crate[RustAnalyzerInfo])
+    if hasattr(ctx.rule.attr, "actual") and ctx.rule.attr.actual != None:
+        dep_infos.append(ctx.rule.attr.actual[RustAnalyzerInfo])
```
Comment (illicitonion) [inline rust/private/rust_analyzer.bzl:69]: This should probably sniff for whether `actual` _has_ a `RustAnalyzerInfo` before trying to read it?

**Review by illicitonion — APPROVED** [review]
I'm not going to lie, this is one of the stranger PRs, but I assume it solves a useful problem :)

File: rust/private/rust_analyzer.bzl (hunk 1 above)
Comment (hlopko) [reply rust/private/rust_analyzer.bzl]: This was a bug :) I wanted to test if there is 'deps' attribute, not a 'proc_macro_deps' attribute. Fixed now.

File: rust/private/rust_analyzer.bzl (hunk 2 above)
Comment (hlopko) [reply rust/private/rust_analyzer.bzl:69]: Done

**Review by UebelAndre — APPROVED** [review]
Thanks!

**Pull request merged**